### PR TITLE
Tons of fixes [delivers #171780075]

### PR DESCRIPTION
**Pull request by betacar**

### CODE REVIEW

File: spec/NCS/client_spec.rb [hunk 1]
```diff
@@ -1,5 +1,45 @@
 require 'spec_helper'
 
 describe NCS::Client do
+  before(:each) { configure_client }
+  let(:subject) { described_class.new(NCS.configuration.to_h) }
```
Comment (dandehavilland) [inline spec/NCS/client_spec.rb]:
```suggestion
  subject { described_class.new(NCS.configuration.to_h) }
```

**Review by dandehavilland — APPROVED** [review]
Could do some moderate refactoring of the client class to remove repetition, but probably not necessary at this stage. 👍

**Comment by betacar** [comment]
@dandehavilland Absolutely agreed. Would love to explode the module, and divide it by scope/domain. Will probably do it soon-ish.

A new review would be appreciated.

**Review by sd — APPROVED** [review]
Looks good to me, assuming it does what it says on the label

**Review by dandehavilland — APPROVED** [review]
☝️what he said 😄

**Pull request merged**